### ISSUE TITLE
Issue adding pool member

### ISSUE THREAD

**Issue by jdwilly2001** [joel74/POSH-LTM-Rest#140]
We have had a working automation that just recently stopped. We are now getting an error while attempting to invoke Add-PoolMember. I do not see any recent commit which may have affected. Any help is appreciated. Will continue plugging away.

```powerhsell
      if($foundPoolMember -eq $false)
	{
        Write-Host "Unable to find pool member $poolMemberName. Adding with status '$status'..."
        Write-Host "Pool Name: $poolName"
        Write-Host "Pool Member Name: $poolMemberName"
        Write-Host "IP Address: $poolMemberIPAddress"
        Write-Host "Port: $poolMemberPortNumber"
        Write-Host "Session: $session"
        Write-Host "Status: $status"
        Add-PoolMember -F5Session $session -PoolName $poolName -Name $poolMemberName -Address $poolMemberIPAddress -PortNumber $poolMemberPortNumber -Status $status 
		Write-Host "Added pool member '$poolMemberName'."
	}
```

This is producing the output:
```
Module Version: 1.4.185
Connected to F5 LTM: 34.202.38.123
--------------------------------------------
...
Unable to find pool member 172.17.45.144:9333. Adding with status 'Disabled'...
Pool Name: pl_widgets.dev.lendingtree.com
Pool Member Name: 172.17.45.144:9333
IP Address: 172.17.45.144
Port: 9333
Session: @{Name=34.202.38.123; BaseURL=https://34.202.38.123/mgmt/tm/ltm/; Credential=; WebSession=Microsoft.PowerShell.Commands.WebRequestSession; LTMVersion=13.0.0}
Status: Disabled
Where-Object : Exception calling "IsMatch" with "2" argument(s): "No such host is known"
At C:\Program Files\WindowsPowerShell\Modules\F5-LTM\1.4.185\Public\Get-Node.ps1:37 char:17
+ ...             Where-Object { [PoshLTM.F5Address]::IsMatch($itemaddress, ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Where-Object], MethodInvocationException
    + FullyQualifiedErrorId : SocketException,Microsoft.PowerShell.Commands.WhereObjectCommand
The remote script failed with exit code 1
```

**Comment by joel74** [comment]
Hi, I tried this with both the latest version of the module (202) and the version you mentioned (185) on LTM v16.1 and I couldn't repro the issue. What LTM version are you working with?
Thanks.

**Comment by jdwilly2001** [comment]
@joel74  I am using it with v13.0 of the LTM.

**Comment by joel74** [comment]
Thanks, @jdwilly2001. I should have caught that on the right side of your code sample. I tested this morning on 13.0 HF2 without issues. It seems like Get-Node isn't happy, presumably either with the value of  $itemaddress or with the $JSON it's trying to null-coalesce. If you can add lines to the Get-Node.ps1 file to write out the values of these two variables, that may provide us with a lead.
Thanks,
Joel

**Comment by joel74** [comment]
Hi, @jdwilly2001 , is the node you're trying to add a FQDN node perchance?

**Comment by jdwilly2001** [comment]
@joel74  no, it is not a FQDN node. We added FQDN node to the system, testing it out, and added a DNS resolver, but removed the hosts and the DNS resolver after we started to see issues.

About your ealier comment I will add some additional debug when i get back to it. We had to revert the script to invoke the iControl REST directly. Probably on Monday

**Comment by joel74** [comment]
Hi @jdwilly2001, I'm just checking in to see if you've had a chance to collect this debug data.

Thanks
Joel

**Comment by vercellone** [comment]
Perhaps a Test-PooMember function should be added.

**Comment by joel74** [comment]
Hi, @jdwilly2001, I believe I've resolved this issue in this commit https://github.com/joel74/POSH-LTM-Rest/commit/1730ad3649b9ac583edcab6acd6c1e90a194f9c2. Get-, Add- and Remove-PoolMember functions now work as expected when working with both FQDN and non-FQDN nodes in BIG-IP versions 11.5, 12.1 and 13.0. If you have a chance and can confirm that this commit fixes your issue in the next several days before I merge the changes into the master branch, I'd appreciate it.
Thanks,
Joel

**Comment by joel74** [comment]
Fixed with PR #144
